exporter/exporterhelper/common.go

```diff
@@ -132,6 +132,15 @@ func WithCapabilities(capabilities consumer.Capabilities) Option {
 	}
 }
 
+func WithBatcher(batcher *Batcher) Option {
+	return func(o *baseExporter) {
+		if !o.requestExporter {
+			panic("batching is only available for the new request exporters")
```

Comment:
Instead of panicking we could add a private field for errors that we can check and use to report errors through `newBaseExporter`